XenAdmin/Dialogs/OptionsPages/ExternalToolsOptionsPage.cs

```diff
@@ -40,54 +40,96 @@ namespace XenAdmin.Dialogs.OptionsPages
     public partial class ExternalToolsOptionsPage : UserControl, IOptionsPage
     {
         private TextBox _tooltipControl;
+        private bool _updating;
+
         public ExternalToolsOptionsPage()
         {
             InitializeComponent();
             sshConsoleInfoLabel.Text = string.Format(sshConsoleInfoLabel.Text, BrandManager.BrandConsole);
-
-            var customSshConsole = Properties.Settings.Default.CustomSshConsole;
-
-            radioButtonOpenSsh.Checked = customSshConsole == SshConsole.OpenSSH;
-            radioButtonPutty.Checked = customSshConsole == SshConsole.Putty;
         }
 
         #region IOptionsPage Members
 
         public void Build()
         {
+            try
+            {
+                _updating = true;
+                textBoxPutty.Text = Properties.Settings.Default.PuttyLocation;
+                textBoxOpenSsh.Text = Properties.Settings.Default.OpenSSHLocation;
+            }
+            finally
+            {
+                _updating = false;
+            }
 
+            var customSshConsole = Properties.Settings.Default.CustomSshConsole;
+
+            if (customSshConsole == SshConsole.Putty)
+                radioButtonPutty.Checked = true;
+            else if (customSshConsole == SshConsole.OpenSSH)
+                radioButtonOpenSsh.Checked = true;
         }
 
         public bool IsValidToSave()
         {
-            if (radioButtonPutty.Checked && !File.Exists(textBoxPutty.Text))
+            _tooltipControl = null;
+
+            // CA-362355: don't check if paths are valid if they haven't changed.

```

Comment:
This comment references a ticket number is it still relevant?

Reply:
I thought I'd add it for more context, as it's not entirely clear as to why this check is here